ocean_lib/ocean/ocean.py

```diff
@@ -216,11 +219,76 @@ def get_user_orders(
 
         return _orders
 
+    @property
+    @enforce_types
+    def fixed_rate_exchange(self):
+        return FixedRateExchange(
+            self.web3, get_address_of_type(self.config, "FixedPrice")
+        )
+
+    @enforce_types
+    def create_fixed_rate(
+        self,
+        erc20_token: ERC20Token,
+        base_token: ERC20Token,
+        amount: int,
+        from_wallet: Wallet,
+    ) -> bytes:
+        fixed_price_address = get_address_of_type(self.config, "FixedPrice")
+        erc20_token.approve(fixed_price_address, amount, from_wallet)
+        addresses = [
+            self.OCEAN_address,
+            from_wallet.address,
+            from_wallet.address,
+            ZERO_ADDRESS,
+        ]
+        uints = [erc20_token.decimals(), base_token.decimals(), to_wei(1), int(1e15), 0]
+
+        fixed_rate_data = FixedData(
+            fixed_price_address=fixed_price_address,
+            addresses=addresses,
+            uints=uints,
+        )
+
+        tx = erc20_token.create_fixed_rate(
+            fixed_data=fixed_rate_data, from_wallet=from_wallet
+        )
+        tx_receipt = self.web3.eth.wait_for_transaction_receipt(tx)
+        fixed_rate_event = erc20_token.get_event_log(
+            ERC721FactoryContract.EVENT_NEW_FIXED_RATE,
+            tx_receipt.blockNumber,
+            self.web3.eth.block_number,
+            None,
+        )
+        exchange_id = fixed_rate_event[0].args.exchangeId
+
+        return exchange_id
+
+    @enforce_types
+    def search_exchange_by_datatoken(self, datatoken: str) -> List[AttributeDict]:
```

Comment:
I would suggest using the function from the nft_factory directly and move the log retrieval itself over there. No point in this being in Ocean class, since there's not much use for address abstraction here. So in the readme you can do factory = ocean.get_nft_factory(); factory.search_exchange_by datatoken(datatoken).

Reply:
Sounds fine from my side.